src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringRunListener.java

```diff
@@ -190,12 +192,14 @@ public String get(@Nullable Map<String, String> carrier, String key) {
                 }
             }
         }
+        Attributes attributes = attrBuilders.build();
+        rootSpanBuilder.setAllAttributes(attributes);
 
         // START ROOT SPAN
         Span rootSpan = rootSpanBuilder.startSpan();
         String traceId = rootSpan.getSpanContext().getTraceId();
         String spanId = rootSpan.getSpanContext().getSpanId();
-        MonitoringAction monitoringAction = new MonitoringAction(traceId, spanId);
+        MonitoringAction monitoringAction = new MonitoringAction(traceId, spanId, attributes.asMap());
```

Comment:
This will populate all the attributes, even the ones that are a list of primitives